webui/tests/tests/StepImplementation.ts

```diff
@@ -149,7 +149,9 @@ export default class StepImplementation {
   @Step('Sign in as <username> with <password>')
   public async signInWithPassword(username: string, password: string) {
     await goto(`${BASE_URL}/login`);
-    await t.clear(t.focus(t.$('#login_username')));
+    const loginSelector = t.$('#login_username');
+    await t.focus(loginSelector);
+    await t.clear(loginSelector);
```

Comment:
**question:** once a focus has been achieved, we might be able to just call `await t.clear()` without the selector?

Reply:
yeah it seems so https://docs.taiko.dev/api/clear/ I'll update

Reply:
**suggestion:** I'm trying this out currently on top of your PR and it's been pretty stable so far. I think it addresses the `visibility` issue. wdy think?
```
  @Step('Sign in as <username> with <password>')
  public async signInWithPassword(username: string, password: string) {
    const usernameId = 'login_username';
    const passwordId = 'login_password';
    await goto(`${BASE_URL}/login`);
    await t.textBox({ id: usernameId }).isVisible();
    await t.focus(t.textBox({ id: usernameId }));
    await t.clear();
    await t.write(username);
    if (password !== '') {
      await t.textBox({ id: passwordId }).isVisible();
      await t.focus(t.textBox({ id: passwordId }));
      await t.write(password);
    }
    await clickAndWaitForPage(t.button('Sign In'));
    await t.text(username, t.near(t.$('#avatar'))).exists();
  }
```

Reply:
do you wanna submit that as a separate pr?